torch_xla/core/xla_model.py

```diff
@@ -109,10 +122,17 @@ def get_ordinal(defval=0):
   Returns:
     The replication ordinal of the current thread.
   """
-  if pjrt.using_pjrt():
-    return pjrt.global_ordinal()
+  global g_ordinals
+  global g_thread_id
```

Comment:
I am not an expert on python threading, but wouldn't you want `g_thread_id` to be thread local? I think `global` means it is shared across devices? If my understanding is correct, we also need to make `g_ordinals` to be thread safe since it might be modified by multiple threads? @will-cromar to correct me if I am wrong.

Reply:
You are right, my bad. I might have found a place where we could safely cache these values.

Reply:
ok but then you mentioned dynamo compiles in a different threads... Would it work if we cache in the `_xla_get_default_device_ordinal` instead?

Reply:
Well, the thing is to prevent dynamo graph breaks. So we need to cache it in the topest layer. Otherwise, other things could introduce graph breaks as well.

Reply:
Actually I'm confused. If the threading is implemented in the python layer for V3-8, how do other global variables work?

Reply:
I think they are being shared across threads? Just like two threads will share the same pytorch random seed. A lot of variables is per device(like old device token) so they are OK.

Reply:
This threading support in V3 is very confusing. We might better to not support Dynamo in V3?

Reply:
yea, I think we don't need to support v3 in this dynamo case now, at least not on PJRT. If there is way to workaround it and not v3 for normal case we should do that.